### PR TITLE
[#IOCIT-110, #ICC-231] third party messages fix

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes
Edited `io-pn-mock` to run the watch script instead of start. This make hot reload available avoiding containers restart.
Added `io-pn-mock` host via traefik
Added `ApiThirdPartyMessageWrite` group to the headers
Changed `THIRD_PARTY_CONFIG_LIST` on `env.io-backend` to ensure a valid value is provided
Fixed PN URLs

#### Motivation and Context
Make the `io-mock` environment ready to test THIRD_PARTY features

#### How Has This Been Tested?
THIRD_PARTY features tested successfully

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: env.io-backend [hunk 1]
```diff
@@ -127,9 +127,9 @@ IS_APPBACKENDLI=true
 # ------------------------------------
 # PN VARIABLES
 # ------------------------------------
-PN_SERVICE_ID=aPNServiceId
-PN_API_URL="http://io-pn-mock:${IO_PN_MOCK_PORT}"
-PN_API_URL_UAT="http://io-pn-mock:${IO_PN_MOCK_PORT}"
+PN_SERVICE_ID=${REQ_SPECIAL_SERVICE_ID}
+PN_API_URL="http://host.docker.internal:${IO_PN_MOCK_PORT}"
```
Comment (michaeldisaro) [inline env.io-backend:131]: isn't this useless if we use traefik?

**Review by michaeldisaro — COMMENTED** [review]
ok

File: env.io-backend (hunk 1 above)
Comment (michaeldisaro) [reply env.io-backend:131]: I mean: using docker internal host.

Comment (arcogabbo) [reply env.io-backend:131]: i will revert this back to the container name, i think it is more clear

**Review by michaeldisaro — APPROVED** [review]

**Review by AleDore — APPROVED** [review]

**Pull request merged**